src/posthog-core.ts

```diff
@@ -349,6 +350,14 @@ export class PostHog {
             })
         )
 
+        // precompile the regex
+        if (!!this.config?.event_capture_config?.url_allowlists) {
+            const url_allowlists = this.config.event_capture_config.url_allowlists
+            Object.entries(url_allowlists).forEach(([key, value]) => {
+                url_allowlists[key] = value.map((url) => new RegExp(url))
```

Comment:
Regexps are dangerous. You can't just pass random unsanitised strings into them.